samples: basic: blinky_pwm: Add support for nucleo_wba55cg

lkoudelka opened the pull request:
Added configuration to the nucleo_wba55cg overlay to enable support for the blinky_pwm demo. This includes the configuration of timers3 and pwm3 nodes.

gmarull reviewed (changes requested):
why isn't this done at board level?

FRASTM commented on samples/basic/blinky_pwm/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,25 @@
+
+
+/ {
+	pwmleds: pwmleds {
```
yes, put node and alias in the board DTS

kartben replied:
What do you mean by complicated? It is a board feature, so it makes sense that it's defined at the board level for anyone to readily use. It's done that way already for literally dozens of Nucleo and Disco boards :)

FRASTM commented on samples/basic/blinky_pwm/boards/nucleo_wba55cg.overlay:
```diff
@@ -0,0 +1,25 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2022 STMicroelectronics
```
new file with new Copyright date

kartben commented on samples/basic/blinky_pwm/README.rst:
```diff
@@ -53,6 +53,8 @@ In these other cases, however, manual wiring is necessary:
      - connect PWM1 (PA8) to an LED
    * - :zephyr:board:`nucleo_wb55rg`
      - connect PWM1 (PA8) to an LED
+   * - :zephyr:board:`nucleo_wba55cg`
+     - connect PWM2 (PA9) to an LED
```
I'm confused, isn't the LED already populated on the board, hence why you were asked to move its definition to the board devicetree? Or is there no such LED on the board, in which case that request was wrong? :/

lkoudelka committed:
boards: nucleo_wba55cg: Add timer3 and pwmleds configuration

Added timer3 and pwmleds configuration to the nucleo_wba55cg board DTS
file. This includes setting the prescaler, enabling the pwm3 node with the
appropriate pin control settings, and configuring the pwmleds node with
green_pwm_led.

Signed-off-by: Lubos Koudelka <lubos.koudelka@st.com>

erwango commented:
@gmarull, @kartben proposed changes look better now. PTAL

rustyh1 reviewed (approved):
Sounds good for me. I have successful tested the change with the nucleo_wba55cg board

gautierg-st commented:
> @erwango needs your approval too :)

@kartben, FYI Erwan is OoO this week ;)
